motoman_driver/src/simple_message/motoman_read_single_io_reply.cpp

```diff
@@ -80,9 +80,17 @@ std::string ReadSingleIOReply::getResultString(shared_int result_code)
 {
   switch (result_code)
   {
-  case ReadSingleIOReplyResults::FAILURE:
-    return "Failed";
-  case ReadSingleIOReplyResults::SUCCESS:
+  case ReadSingleIOReplyResultCodes::READ_ADDRESS_INVALID:
+     return "Illegal address for read: outside permitted range on this controller, see documentation (1001)";
+  case ReadSingleIOReplyResultCodes::WRITE_ADDRESS_INVALID:
+     return "Illegal address for write: outside permitted range on this controller, see documentation (1002)";
+  case ReadSingleIOReplyResultCodes::WRITE_VALUE_INVALID:
+     return "Illegal value for the type of IO element addressed (1003)";
+  case ReadSingleIOReplyResultCodes::READ_API_ERROR:
+     return "The MotoPlus function MpReadIO returned -1. No further information is available (1004)";
+  case ReadSingleIOReplyResultCodes::WRITE_API_ERROR:
+     return "The MotoPlus function MpWriteIO returned -1. No further information is available (1005)";

```

Comment:
It might be worth changing the strings to use the value directly from the enumeration rather than hard-coding the numbers, just in case they change in the future, i.e.:

```c++
return "Illegal address for read: outside permitted range on this controller, see documentation (" + std::to_string(ReadSingleIoReplyResultCodes::READ_ADDRESS_INVALID) + ")";
```

Reply:
I'll see if I change this. We're not really using C++11 anywhere in this codebase, this would be a divergence, but that could be OK.

Reply:
Decided it was worth it. Driver now builds with C++11 enabled, so we get to use `std::to_string(..)`.

thanks for the suggestion.